### DISCUSSION TITLE
Protecting against injection attacks

### DISCUSSION THREAD

**Discussion by wz2b** [alexandrainst/node-red-contrib-postgresql#60] [Q&A]
Do [Numeric Parameter](https://github.com/alexandrainst/node-red-contrib-postgresql/blob/main/README.md#parameterized-query-numeric) queries use an underlying parameterized query mechanism, or is it just a string replacement?  I'm trying to understand how to protect against SQL injection attacks.  Using moustache templates to just assemble a query string is [very](https://owasp.org/www-community/attacks/SQL_Injection) [very](https://www.w3schools.com/sql/sql_injection.asp) [dangerous](https://medium.com/@V-Blaze/sql-injection-protecting-your-postgresql-database-ce8c0cc43685)

**Comment by wz2b** [comment]
I think I found the answer [here](https://node-postgres.com/features/queries)

I would like to suggest the documentation should be updated to say that while this library supports moustache templates it's not recommended for security reasons.

**Reply by christian-skjetne** [reply]
I would agree that a (big fat) warning in the documentation would be a good idea. Parameterized queries should be safe, but using moustache templates makes injection trivial if it is based on user input.

**Comment by wz2b** [comment]
I believe from reading the code that named parameters are "emulated" by converting them into numbered parameters, meaning that this mechanism also is safe against injection - is that correct?  It would be good if the documentation said this.  If someone with knowledge fot his can confirm that's true I'll update the docs and submit a PR, someone let me know.

**Comment by Alkarex** [comment]
PR welcome for a documentation update. In any case, sanitising user or third-party input is always a good idea prior further processing.
